ivy/array/layers.py

```diff
@@ -59,3 +59,64 @@ def depthwise_conv2d(
         return ivy.depthwise_conv2d(
             self._data, filters, strides, padding, data_format, dilations, out=out
         )
+
+    def conv1d(
+        self: ivy.Array,
+            filters: Union[ivy.Array, ivy.NativeArray],
+            strides: int,
+            padding: str,
+            data_format: str = "NWC",
+            dilations: int = 1,
+            *,
+            out: Optional[ivy.Array] = None,
+    ) -> ivy.Array:
+        """
+        ivy.Array instance method variant of ivy.conv1d. This method simply
+        wraps the function, and so the docstring for ivy.conv1d also applies
+        to this method with minimal changes.
+
+       Parameters
+    ----------
+    x
+        Input image *[batch_size,w,d_in]*.
+    filters
+        Convolution filters *[fw,d_in,d_out]*.
+    strides
+        The stride of the sliding window for each dimension of input.
+    padding
+        SAME" or "VALID" indicating the algorithm, or list indicating the per-dimension
+        paddings.
+    data_format
+        NWC" or "NCW". Defaults to "NWC".
+    dilations
+        The dilation factor for each dimension of input. (Default value = 1)
+    out
+        optional output array, for writing the result to. It must have a shape that the
+        inputs broadcast to.
+
+    Returns
+    -------
+    ret
+        The result of the convolution operation.
+
+        Examples
+        --------
+        >>> x = ivy.array(ivy.random_normal(0, 1, [2, 12, 4]))  # NWC
+        >>> filters = ivy.array(ivy.random_normal(0, 1, [6, 4, 2]))  # WIO (I == C)
```

Comment:
yeah sure i'll have a look

Reply:
I just reran the example. It worked perfectly fine for me.
Why is it giving an error when you're running?

Reply:
The 2<6 shouldn't be a problem as far as I can tell since the 2 represents the batch-size. As far as the shapes are concerned, The **C** parameter of the input (`NWC`) should match the **I** parameter of the filter(`WIO`).